src/main/java/it/gov/pagopa/xpay/service/XPayPayment3DSService.java

```diff
@@ -0,0 +1,111 @@
+import static it.gov.pagopa.xpay.utils.XPayConstants.XPAY_PAYMENT_OUTCOME;
+
+@Service
+@Log4j2
+public class XPayPayment3DSService {
+    @Value("${xpay.apikey-alias}")
+    private String apiKey;
+
+    @Value("${xpay.secret-key}")
+    private String chiaveSegreta;
+
+    @Autowired
+    private XPayRepository xPayRepository;
+
+    @Autowired
+    private TableConfigRepository configRepository;
+
+    private String outcomeConfig;
+    private XPayErrorEnum errorConfig;
+
+    private void refreshConfigs() {
+        outcomeConfig = configRepository.findByPropertyKey(XPAY_PAYMENT_OUTCOME).getPropertyValue();
+        errorConfig = XPayUtils.getErrorConfig(configRepository.findByPropertyKey(XPAY_PAYMENT_ERROR).getPropertyValue());
+    }
+
+    public ResponseEntity<XPayPaymentResponse> getMock(XPayPaymentRequest request) {
+        log.info("XPay Paga3DS - Request from PGS: " + request);
+        refreshConfigs();
+
+        String codiceTransazione = request.getCodiceTransazione();
+        String idOperazione = UUID.randomUUID().toString();
+        String macToReturn;
+
+        try {
+            log.info("XPay Paga3DS - Generating MAC for transactionId: " + codiceTransazione);
+            macToReturn = XPayUtils.getMacToReturn(codiceTransazione, request.getDivisa(), request.getImporto(),
+                    request.getTimeStamp(), apiKey, chiaveSegreta);
+        } catch (Exception e) {
+            log.error("XPay Paga3DS - Exception during the creation of the MAC string: ", e);
+            XPayErrorEnum error = XPayErrorEnum.ERROR_50;
+
+            return ResponseEntity.status(error.getHttpStatus())
+                    .body(createXPayPaymentResponse(XPayOutcome.KO, idOperazione, null, error));
+        }
+
+        if(outcomeConfig.equals("OK")) {
+            if (macToReturn.equals(request.getMac())) {
+                log.info("XPay Paga3DS - MAC verified");
+                XPayPaymentResponse xPayPaymentResponse = createXPayPaymentResponse(XPayOutcome.OK, idOperazione, macToReturn, null);
+                if(request.getParametriAggiuntivi() != null)
+                    xPayPaymentResponse.setParametriAggiuntivi(request.getParametriAggiuntivi());;
+
+                return ResponseEntity.ok().body(xPayPaymentResponse);
+            } else {
+                log.info("XPay Paga3DS - MAC not verified");
+                XPayErrorEnum error = XPayErrorEnum.ERROR_3;
+
+                return ResponseEntity.status(error.getHttpStatus())
+                        .body(createXPayPaymentResponse(XPayOutcome.KO, idOperazione, macToReturn, error));
+            }
+        } else {
+            return ResponseEntity.status(errorConfig.getHttpStatus())
+                    .body(createXPayPaymentResponse(XPayOutcome.KO, idOperazione, macToReturn, errorConfig));
+        }
+    }
+
+    private XPayPaymentResponse createXPayPaymentResponse(XPayOutcome XPayOutcome, String idOperazione, String mac,
```

Comment:
Not a critical problem, but XPayOutcome parameter name should have a lowercase initial